web/src/main/java/org/cbioportal/web/util/DataBinHelper.java

```diff
@@ -399,4 +405,45 @@ public Boolean areAllIntegers(Set<BigDecimal> uniqueValues) {
             .reduce((a, b) -> a && b)
             .orElse(false);
     }
+
+    public List<BigDecimal> generateBins(List<BigDecimal> sortedNumericalValues, BigDecimal binSize, BigDecimal anchorValue) {
+
+        Assert.notNull(sortedNumericalValues, "sortedNumerical values is null!");
+        Assert.notNull(binSize, "binSize values is null!");
+        Assert.notNull(anchorValue, "anchorValue values is null!");
+
+        BigDecimal minValue = min(sortedNumericalValues);
+        BigDecimal maxValue = max(sortedNumericalValues);
```

Comment:
If these values are sorted, aren't min and max just first and last?

Reply:
Ah yes, I updated the logic, but added checks that the order is as we expect.